### ISSUE TITLE
Build of libzippp[encryption] no longer sets -fPIC on Linux

### ISSUE THREAD

**Issue by BengtGustafsson** [microsoft/vcpkg#30983]
**Describe the bug**
In the tag 2023.02.24 the libzippp[encryption] build for x64-linux no longer sets the -fPIC (position independent code) when compiling which makes it impossible to link to this library in our application.

This worked before the 2023.01.09 tag sometime during the fall of 2022, but not in that tag.

**Environment**
- OS: linux ubuntu 18.04
- Compiler: gcc 11.2

**To Reproduce**
1. ./vcpkg install  libzippp[encryption] 
2. See error
Link the resulting library into an executable or .so file.

**Expected behavior**
It works

**Failure logs**
Linker error from ld:
source/externals/libs/installed/x64-linux/lib/libzippp_static.a(libzippp.cpp.o): relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC

**Additional context**

**Comment by MonicaLiu0311** [comment]
vcpkg version:
```
monica@monica003:~/vcpkg$ ./vcpkg version
vcpkg package management program version 2023-03-29-664f8bb619b752430368d0f30a8289b761f5caba

See LICENSE.txt for license information.
```

usage:
```
libzippp provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(libzippp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE libzip::zip libzippp::libzippp)
```

Here is my build result:
```
monica@monica003:~/CMakeFindUsage/build$ cmake ..
-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/monica/CMakeFindUsage/build
```
```
monica@monica003:~/CMakeFindUsage/build$ cmake --build .
[ 50%] Building CXX object CMakeFiles/main.dir/CMakeFindUsage.cpp.o
[100%] Linking CXX executable main
[100%] Built target main
```

<details>
<summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "libzippp/libzippp.h"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/vcpkg/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")

add_executable (main "CMakeFindUsage.cpp")

find_package(libzippp CONFIG REQUIRED)

target_link_libraries(main PRIVATE libzip::zip libzippp::libzippp)
</pre></details>

**Comment by MonicaLiu0311** [comment]
Please update to the latest version and try.

**Comment by BengtGustafsson** [comment]
Thanks for the update. I'm not positive, but it may be that you actually have to use a symbol in libzippp to trigger the ld error. Please try adding this line to the cpp file:

libzippp::ZipArchive myArhive("foo.zip");

I will try the latest master commit though.

**Comment by BengtGustafsson** [comment]
I managed to compile with 'master' branch of vcpkg as of today, with the same error.

**Comment by MonicaLiu0311** [comment]
> Please try adding this line to the cpp file:
> 
> libzippp::ZipArchive myArhive("foo.zip");

I did as follows:
(`testfile.zip` and `CMakeList.txt` are at the same level)

<details><summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;

#include "libzippp/libzippp.h"

using namespace std;

int main()
{
    libzippp::ZipArchive myArhive("testfile.zip");
    cout << "Hello CMake." << endl;
    return 0;
}
<pre/></details>


```
root@monica003:/home/monica/test/build# cmake ..
-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/monica/test/build

root@monica003:/home/monica/test/build# cmake --build .
[ 50%] Building CXX object CMakeFiles/main.dir/test.cpp.o
[100%] Linking CXX executable main
[100%] Built target main

root@monica003:/home/monica/test/build# ./main
Hello CMake.
```

**Comment by BengtGustafsson** [comment]
I took the libzippp.cpp file out from the vcpkg source tree and compiled it with our application instead. Then it worked, so there is definitely something fishy about how vcpkg compiles it. I blamed the CMakeLists.txt file in the libzippp master repo and there were changes made 3-4 months ago which is consistent with when the vcpkg tags stopped working, so maybe you bumped the libzippp version vcpkg provides around this time and there was a problem introduced by this.

As libzippp is just one cpp file we can live with our work around for now, but it is of course a hack so please try to figure out what happened.

**Comment by BengtGustafsson** [comment]
I found a closed issue in the libzippp repo: [https://github.com/ctabin/libzippp/issues/158](https://github.com/ctabin/libzippp/issues/158)

NOTE: The reason you can't reproduce may be that you create an executable, try changing that to a shared library;

add_library(testLib SHARED CMakeFindUsage.cpp)

**Comment by dg0yt** [comment]
The issue is there:
https://github.com/ctabin/libzippp/blob/ec69c0d2b2d46877b67852e27a485f7e754b6f78/CMakeLists.txt#L27-L31

This overwrite flags from the toolchain.

**Comment by ctabin** [comment]
This should have been [fixed](https://github.com/ctabin/libzippp/pull/194) in the release 6.1-1.9.2.

**Comment by MonicaLiu0311** [comment]
Fixed by #31194.
